OpenRA.Test/OpenRA.Game/SyncImplTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+using OpenRA;
+
+namespace OpenRA.Test
+{
+	[TestFixture]
+	class SyncImplTest
+	{
+		static readonly IEnumerable<Type> AllModTypes =
+			Directory.EnumerateFiles("mods", "OpenRA.Mods.*.dll", SearchOption.AllDirectories)
+			.Select(path => Assembly.LoadFile(Path.GetFullPath(path)))
+			.SelectMany(assembly => assembly.GetTypes());
+
+		static readonly Type SyncInterface = typeof(ISync);
+
+		static bool TypeImplementsSync(Type type)
+		{
+			return type.GetInterfaces().Contains(SyncInterface);
+		}
+
+		static bool AnyTypeMemberIsSynced(Type type)
+		{
+			const BindingFlags Flags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly;
+			while (type != null)
+			{
+				if (((MemberInfo[])type.GetFields(Flags)).Concat(type.GetProperties(Flags)).Any(x => x.HasAttribute<SyncAttribute>()))
+					return true;
+				type = type.BaseType;
+			}
+
+			return false;
+		}
+
+		[TestCase(TestName = "Types with syncable members implement sync interface")]
+		public void TypesWithSyncableMembersImplementSyncInterface()
+		{
+			foreach (var type in AllModTypes)
+				if (!TypeImplementsSync(type) && AnyTypeMemberIsSynced(type))
+					Assert.Fail("{0} has members with the Sync attribute but does not implement ISync".F(type.FullName));
+			Assert.Pass();
```

Comment:
```
at NUnit.Framework.Assert.Pass(String message, Object[] args)
at NUnit.Framework.Assert.Pass()
at OpenRA.Test.SyncImplTest.TypesImplementingSyncInterfaceHaveSyncableMembers() in c:\projects\openra\OpenRA.Test\OpenRA.Game\SyncImplTest.cs:line 64
```

What does this mean? Does it think Assert.Pass() indicates a failure or something?